lib/nostrum/api.ex

```diff
@@ -138,51 +138,108 @@ defmodule Nostrum.Api do
+
+  Nostrum.Api.create_message(43189401384091, content: "hello world!", embed: embed, file: "/path/to/file.txt")
   ```
   """
-  @spec create_message(Message.t(), message_content, boolean) :: error | {:ok, Message.t()}
-  @spec create_message(Channel.id(), message_content, boolean) :: error | {:ok, Message.t()}
-  def create_message(channel_id, content, tts \\ false)
+  @spec create_message(Channel.id(), options) :: error | {:ok, Message.t()}
+  def create_message(channel_id, options)
 
-  def create_message(%Message{channel_id: id}, content, tts) when is_binary(content),
-    do: create_message(id, content, tts)
+  def create_message(channel_id, options) when is_list(options),
+    do: create_message(channel_id, Map.new(options))
 
-  # Sending regular messages
-  def create_message(channel_id, content, tts) when is_binary(content),
-    do: do_create_message(channel_id, %{content: content, tts: tts})
+  def create_message(channel_id, options) when is_snowflake(channel_id) do
+    case options do
+      %{file: _} -> create_message_with_multipart(channel_id, options)
+      %{} -> create_message_with_json(channel_id, options)
+    end
+  end
 
-  # Embeds
-  def create_message(channel_id, [content: c, embed: e], tts),
-    do: do_create_message(channel_id, %{content: c, embed: e, tts: tts})
+  defp create_message_with_multipart(channel_id, %{file: file_path} = options) do
+    payload_json =
+      options
+      |> Map.delete(:file)
+      |> Poison.encode!()
 
-  # Files
-  def create_message(channel_id, [file_name: c, file: f], tts),
-    do: do_create_message(channel_id, %{content: c, file: f, tts: tts})
+    multipart = [
+      {:file, file_path},
+      {"payload_json", payload_json}
+    ]
 
-  @doc """
-  Send a message to a channel.
+    request = %{
+      method: :post,
+      route: Constants.channel_messages(channel_id),
+      body: {:multipart, multipart},
+      options: [],
+      headers: [
+        {"content-type", "multipart/form-data"}
+      ]
+    }
 
-  See `create_message/3` for usage.
+    GenServer.call(Ratelimiter, {:queue, request, nil}, :infinity)
+    |> handle_request_with_decode({:struct, Message})
+  end
 
-  Raises `Nostrum.Error.ApiError` if error occurs while making the rest call.
+  defp create_message_with_json(channel_id, options) do
+    request(:post, Constants.channel_messages(channel_id), options)
+    |> handle_request_with_decode({:struct, Message})
+  end
+
+  @doc ~S"""
+  Same as `create_message/2`, but raises `Nostrum.Error.ApiError` in case of failure.
+  """
+  @spec create_message!(Channel.id(), options) :: no_return | Message.t()
+  def create_message!(channel_id, options) do
+    create_message(channel_id, options)
+    |> bangify
+  end
+
+  @doc ~S"""
+  Same as `create_message/2`, but takes a `Nostrum.Struct.Message` instead of a channel_id.
+  """
+  @spec reply_to_message(Message.t(), options) :: error | {:ok, Message.t()}
+  def reply_to_message(%Message{} = message, options) do
```

Comment:
I don't think I like having a separate method for passing in a message struct. I can understand it's probably this way since right now we're guarding against the list and we don't want to have to deal with the combinations. However, given the choice I think it's probably better to match on the message struct and handle the list conversion inside the function.